server/api.ts

```diff
@@ -1357,6 +1370,17 @@ router.getAsync("/teams", async (req, res) => {
  */
 router.deleteAsync("/team", ensure.isAdmin, async (req, res) => {
   logger.info(`${req.user.username} deleted team ${req.body._id} from ${req.body.tourney}`);
+  // Unregister the players as well if the tourney has registerAsTeam flag enabled
+  const tourney = await Tournament.findOne({ code: req.body.tourney }).orFail();
+  if (tourney.flags.includes("registerAsTeam")) {
+    const team = await Team.findOne({ _id: req.body._id });
+    team!.players.forEach(async (player) => {
+      await User.findOneAndUpdate(
+        { _id: player._id },
+        { $pull: { tournies: req.body.tourney, stats: { tourney: req.body.tourney } } }
+      );
+    });
+  }
```

Comment:
to await these like in the register team route id probably do smth like
```
const team = await Team.findOne({ _id: req.body._id }).orFail();
await Promise.all(
  team.players.map((player) => 
    User.findOneAndUpdate(
      { _id: player._id },
      { $pull: { tournies: req.body.tourney, stats: { tourney: req.body.tourney } } }
    )
  )
);
```

Reply:
👍